lizmap/modules/admin_api/lib/RepoCreator.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace LizmapApi;
+
+class RepoCreator
+{
+    /**
+     * Creates a new repository with the given properties.
+     *
+     * @param string $key                    The unique identifier for the repository. Must not yet exist.
+     * @param string $label                  the display name of the repository
+     * @param string $path                   the path to the repository directory
+     * @param string $allowUserDefinedThemes whether user-defined themes are allowed for the repository
+     *
+     * @return bool returns true if the repository is successfully created
+     *
+     * @throws \Exception if the repository creation fails
+     */
+    public static function createRepository(string $key, string $label, string $path, string $allowUserDefinedThemes): bool
+    {
+        if (\lizmap::getRepository($key)) {
+            throw new \Exception("The repository '{$key}' already exists.", 409);
```

Comment:
I invite you to defined an `ApiException` that extends `\Exception`, just like `UnknownLizmapProjectException`:

```php
<?php

namespace LizmapApi;

class ApiException extends \Exception {}

```

Then when you catch it you know that the code from `getCode()` is an HTTP code.